build-logic/src/main/kotlin/config-shade.gradle.kts

```diff
@@ -0,0 +1,41 @@
+import com.github.jengelman.gradle.plugins.shadow.tasks.ShadowJar
+import gradle.kotlin.dsl.accessors._d73e0aad27c892ff134862f94ec3182c.build
+import gradle.kotlin.dsl.accessors._d73e0aad27c892ff134862f94ec3182c.implementation
+import gradle.kotlin.dsl.accessors._d73e0aad27c892ff134862f94ec3182c.jar
+import org.gradle.kotlin.dsl.creating
+
+plugins {
+    id("com.github.johnrengelman.shadow")
+}
+
+val shade: Configuration by configurations.creating
+configurations.implementation { extendsFrom(shade) }
+
+tasks {
+    val shadowJar by getting(ShadowJar::class) {
+        archiveClassifier.set("all")
+
+        configurations = listOf(shade)
+        exclude("META-INF/*.SF", "META-INF/*.DSA", "META-INF/*.RSA", "OSGI-INF/**", "*.profile", "module-info.class", "ant_tasks/**")
+
+        arrayOf(
+            "com.google",
+            "org.objectweb.asm",
+            "org.spongepowered",
+            "com.grappenmaker.mappings",
+        ).forEach { pkg ->
```

Comment:
Unwanted string relocation is fixed with latest commits, all that remains is understanding why gson/guava can't be relocated, because I see no issues with them in terms of relocated code.

Reply:
Appears to be yet another dead code thing, in an earlier branch there was a need to do so as a result of improperly setting up relocation for Mixin.